browser/caliban.py

```diff
@@ -205,12 +195,17 @@ def action_trim_pixels(self, label, frame, x_location, y_location):
         img_ann = self.annotated[frame,:,:,self.feature]
         contig_cell = flood(image = img_ann, seed_point = (int(y_location/self.scale_factor), int(x_location/self.scale_factor)))
         img_trimmed = np.where(np.logical_and(np.invert(contig_cell), img_ann == label), 0, img_ann)
-        self.annotated[frame,:,:,self.feature] = img_trimmed
 
+        #check if image changed
+        comparison = np.where(img_trimmed != img_ann)
+        self.frames_changed = np.any(comparison)
+        #this action should never change the cell info
+
+        self.annotated[frame,:,:,self.feature] = img_trimmed
 
     def action_handle_draw(self, trace, edit_value, brush_size, erase, frame):
 
-        annotated = self.annotated[frame,:,:,self.feature]
+        annotated = np.copy(self.annotated[frame,:,:,self.feature])
```

Comment:
Verify this copy call behaves as expected. ie: shallow copy vs deep copy

Reply:
same for line 728

Reply:
np.copy works analogously to deep copy in this case (when annotated is modified, self.annotated is not modified)
https://docs.scipy.org/doc/numpy-1.16.0/reference/generated/numpy.copy.html